test/test_matmul_scheduler.cpp

```diff
@@ -15,16 +15,971 @@
 #include <test/utils.h>
 #include <test/validator.h>
+#include "ops/arith.h"
+#include "type.h"
 
 namespace nvfuser {
 
 namespace {
 class MatmulSchedulerTest : public NVFuserTest {};
+
+using PrecisionsDesc = std::pair<PrimDataType, PrimDataType>;
+
+using AbsoluteError = double;
+using RelariveError = double;
+using ErrorThresholds = std::pair<AbsoluteError, RelariveError>;
+using TestCaseErrorThresholds = std::map<PrecisionsDesc, ErrorThresholds>;
+class PrecisionParametrizedTest
+    : public NVFuserFixtureParamTest<PrecisionsDesc> {};
+
+[[nodiscard]] auto getAtType(const PrimDataType& type) {
+  switch (type) {
+    case PrimDataType::Half:
+      return at::kHalf;
+    case PrimDataType::Float:
+      return at::kFloat;
+    case PrimDataType::BFloat16:
+      return at::kBFloat16;
+    default:
+      break;
+  }
+  NVF_ERROR(false, "Unsupported conversion of PrimDataType");
+  return at::kHalf;
+}
+
+static const PrecisionsDesc HSH =
+    std::make_pair(PrimDataType::Half, PrimDataType::Half);
+static const PrecisionsDesc HSS =
+    std::make_pair(PrimDataType::Half, PrimDataType::Float);
+static const PrecisionsDesc TST =
+    std::make_pair(PrimDataType::BFloat16, PrimDataType::BFloat16);
+static const PrecisionsDesc TSS =
+    std::make_pair(PrimDataType::BFloat16, PrimDataType::Float);
+
+// Matmul test that uses segmenter for fusion:
+//   D = (A x B) + bias
+//  Target architectures: Turing, Ampere
+TEST_P(PrecisionParametrizedTest, EpilogueBias) {
+  NVFUSER_TEST_CUDA_ARCH_RANGE_GUARD(7, 5, 9, 0);
+  const auto layout = MatmulLayout::TT;
+
+  static TestCaseErrorThresholds errs = {
+      {HSS, std::make_pair(0.0001, 0.0001)},
+      {HSH, std::make_pair(0.001, 0.001)},
+      {TSS, std::make_pair(0.0001, 0.0001)},
+      {TST, std::make_pair(0.01, 0.01)},
+  };
+
+  NVF_CHECK(
+      errs.count(GetParam()) != 0,
+      "Undefined error thresholds for requested precisions");
+
+  const auto [in_prim_type, out_prim_type] = GetParam();
+  const auto [abs_err_thr, rel_err_thr] = errs[GetParam()];
+
+  // NOTE: default precisions in (fp16) -> compute(fp32) -> out(fp32)
+  const auto default_out_type = (PrimDataType::Float == out_prim_type);
+  const auto in_type = DataType(in_prim_type);
+  const auto out_type = DataType(out_prim_type);
+  const auto at_in_type = getAtType(in_prim_type);
+  const auto at_out_type = getAtType(out_prim_type);
+
+  // NOTE: bfloat16 is not supported on pre-Ampere archs
+  if (DataType::BFloat16 == in_type || DataType::BFloat16 == out_type) {
+    NVFUSER_TEST_CUDA_ARCH_GUARD(8, 0);
+  }
+
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  // A - tv0, B - tv1, C - tv2
+  auto tv0 = makeContigTensor(2, in_type);
+  auto tv1 = makeContigTensor(2, in_type);
+  auto tv2 = makeContigTensor(1, out_type);
+
+  // tv3 := A x B
+  auto tv3 = matmul(tv0, tv1, layout, true);
+  // tv4 := cast(tv3)
+  auto tv4 = default_out_type ? tv3 : castOp(out_type, tv3);
```

Comment:
Does this work?
https://github.com/NVIDIA/Fuser/blob/e07e569354a57166cf66ba9ccf427f6f2ebd57ff/csrc/ops/arith.h#L32-L33

Reply:
This was the tool I was missing, thanks!